src/compiler/diagnosticMessages.json

```diff
@@ -3310,6 +3310,10 @@
         "category": "Error",
         "code": 7034
     },
+    "Try `npm install @types/{0}` if it exists or adding `declare module '{0}'` in a separate file.": {

```

Comment:
Technically, any old .ts file is fine. @andy-ms do we provide any guidance on the *right* way to do this?

For "in your compilation", maybe reword the last half to say "... or add a new file [to your project] containing `declare module 'foo'`." Sound better? I'm not sure about 'to your project'.